src/minmax.rs

```diff
@@ -0,0 +1,90 @@
+use rustc::lint::{Context, LintPass, LintArray};
+use rustc_front::hir::*;
+use syntax::codemap::Spanned;
+use syntax::ptr::P;
+use std::cmp::PartialOrd;
+use std::cmp::Ordering::*;
+
+use consts::{Constant, constant};
+use utils::{match_path, span_lint};
+use self::MinMax::{Min, Max};
+
+declare_lint!(pub MIN_MAX, Deny,
```

Comment:
So it's about the complexity of the lint – you don't trust the math being correct or the code correctly implementing the math in all cases, right? Perhaps you're right in the case of `bad_bit_mask`, but I'd rather calm your fears by adding more comprehensive tests than reducing the lint level, because *if* the lint is right, there is no way the reported code could have been written deliberately, unless someone is being sneaky or trying to test clippy :smile:.